src/pyhf/cli/infer.py

```diff
@@ -17,6 +18,119 @@ def cli():
     """Infererence CLI group."""
 
 
+@cli.command()
+@click.argument("workspace", default="-")
+@click.option(
+    "--output-file",
+    help="The location of the output json file. If not specified, prints to screen.",
+    default=None,
+)
+@click.option("--measurement", default=None)
+@click.option("-p", "--patch", multiple=True)
+@click.option(
+    "--value",
+    help="Flag for returning the fitted value of the objective function.",
+    default=False,
+    is_flag=True,
+)
+@click.option(
+    "--backend",
+    type=click.Choice(["numpy", "pytorch", "tensorflow", "jax", "np", "torch", "tf"]),
+    help="The tensor backend used for the calculation.",
+    default="numpy",
+)
+@click.option(
+    "--optimizer",
+    type=click.Choice(["scipy", "minuit"]),
+    help="The optimizer used for the calculation.",
+    default="scipy",
+)
+@click.option("--optconf", type=EqDelimStringParamType(), multiple=True)
+def fit(
+    workspace,
+    output_file,
+    measurement,
+    patch,
+    value,
+    backend,
+    optimizer,
+    optconf,
+):
+    """
+    Perform a maximum likelihood fit for a given pyhf workspace.
+
+    Example:
+
+    .. code-block:: shell
+
+        $ curl -sL https://git.io/JJYDE | pyhf fit --value
+
+        \b
+        {
+            "mle_parameters": {
+                "mu": [
+                    0.00017298628839781602
+                ],
+                "uncorr_bkguncrt": [
+                    1.0000015671710816,
+                    0.9999665895859197
+                ]
+            },
+            "twice_nll": 23.19636590468879
```

Comment:
should we return here NLL given #1057 ?

Reply:
ok see the other comment..

Reply:
To be clear, I'm fine with getting PR #1057 in first and then this one. I'm just waiting on further feedback on it.